src/infrastructure/noteSettings.repository.ts

```diff
@@ -0,0 +1,65 @@
+import type NoteSettingsRepositoryInterface from '@/domain/noteSettings.repository.interface';
+import type NotesSettings from '@/domain/entities/NotesSettings';
+import type NoteStorage from '@/infrastructure/storage/note';
+import type NotesApiTransport from '@/infrastructure/transport/notes-api';
+import type { GetNotesSettingsResponsePayload } from '@/infrastructure/transport/notes-api/types/GetNoteResponsePayload';
+
+/**
+ * Note repository
+ */
+export default class NoteSettingsRepository implements NoteSettingsRepositoryInterface {
+  /**
+   * Transport instance
+   */
+  private readonly transport: NotesApiTransport;
+
+  /**
+   * Note storage
+   */
+  private noteStorage: NoteStorage;
+
+  /**
+   * Note repository constructor
+   *
+   * @param noteStorage - note storage instance
+   * @param notesApiTransport - notes api transport instance
+   */
+  constructor(noteStorage: NoteStorage, notesApiTransport: NotesApiTransport) {
+    this.noteStorage = noteStorage;
+    this.transport = notesApiTransport;
+  }
+
+  /**
+   * Get NotesSettings by note ID
+   *
+   * @param publicId - Note publicId
+   * @returns { NotesSettings | null } - NotesSettings instance
+   */
+  public async getNotesSettingsById(publicId: string): Promise<NotesSettings | null> {
+    const notesSettingsData = await this.noteStorage.getNotesSettingsById(publicId);
+
+    /**
+     * If notesSettings data in storage exists
+     */
+    if (notesSettingsData) {
+      return notesSettingsData;
+    }
```

Comment:
these lines have no sense, since we have removed noteSettings storage

Reply:
Because there should be a clear reason to add a cache, and for now we don't have that reason.